### ISSUE TITLE
Please add a "Solo" layer option

### ISSUE THREAD

**Issue by f1nalspace** [mapeditor/tiled#835]
It would be incredible useful if tiled would support solo-layering (Make ony single layer editable only as needed - all others but readable). An icon next to the checkbox with a dot indicates when a layer is "solo". Double click on layer will set/toggle solo - like in photoshop.

**Comment by ketanhwr** [comment]
Should this be closed? Or maybe we can add a key combination which would lock all the layers apart from the current layer.

**Comment by bjorn** [comment]
> Should this be closed? Or maybe we can add a key combination which would lock all the layers apart from the current layer.

Right, maybe we could use Ctrl+Shift+L.

I wonder if it would make sense to also copy the double-click behavior from Photoshop, though I'm not entirely sure how that works. In Tiled a double-click will edit the layer name. Maybe we could handle double-click on the lock icon.

**Comment by olatan** [comment]
Really want to second this, a simple solo layer switch next to the lock/visibility switches would be amazing for workflow speed. if(any soloed layers){ show only soloed} else { use visibility }.
Click-dragging should also work on the icons instead of moving the layer in the layer stack like it currently does <3
Really hoping for this feature!

**Comment by bjorn** [comment]
@eishiya helped me to get a bit more context, since I'm not familiar with this feature in Photoshop or After Effects.

What we're after here is an editor-only (not saved) layer visibility state, which can take any of the following values:

* **Default** (use layer's visibility state, but hide when any other layer is Solo)
* **Solo** (this layer is visible, ignoring its visibility state)
* **Mute** (this layer is hidden, ignoring its visibility state)

And the following UI interactions (these are based on PyxelEdit, but are likely similar):

* Click the eye icon to show/hide. Open eye: shown, closed eye: hidden, just like Tiled.
* Alt+click the eye icon to Solo. This puts a frame around the eye icon. Whether hidden or shown, the layer is visible.
* Shift+click the eye icon to Mute, this crosses it out. Whether hidden or shown, the layer is invisible.

![image](https://github.com/mapeditor/tiled/assets/531764/19e427fe-4ef3-4168-b453-b75bb7d9922c)

In the above image, layer 0 is Mute while layers 1 & 2 are Solo. Layers 1 & 2 are visible, even if layer 2 is normally hidden.

**Comment by olatan** [comment]
Exactly, as you say, editor-only layer visibility state. Having a mute state is even a step further than what I was asking, but sounds great! Will it happen?

**Comment by eishiya** [comment]
Just for completion's sake: PyxelEdit also lets you set Solo/Mute via the layer's properties (that Settings gear icon in the screenshot), which is more discoverable than the alt/shift interactions, although those are also mentioned in the tooltip for the eye. In Tiled's case, I think having the Solo/Mute options visible in the right-click menu below the visibility options would be a good idea.
